src/main/parser/Parser.java

```diff
@@ -0,0 +1,158 @@
+package main.parser;
+
+import java.time.DateTimeException;
+import java.time.LocalDateTime;
+
+import main.command.AddDeadlineCommand;
+import main.command.AddEventCommand;
+import main.command.AddTodoCommand;
+import main.command.Command;
+import main.command.DeleteCommand;
+import main.command.DoneCommand;
+import main.command.ExitCommand;
+import main.command.FindCommand;
+import main.command.ListCommand;
+import main.exception.EmptyMessageException;
+import main.exception.InvalidDateException;
+import main.exception.InvalidDeadlineFormatException;
+import main.exception.InvalidEventFormatException;
+import main.exception.InvalidTaskException;
+import main.exception.UnknownCommandException;
+
+/**
+ * Handles the parsing of user inputs.
+ * @author Joshua Liang XingYa
+ * @author joshualiang.xy@gmail.com
+ * @version v0.1
+ * @since v0.1
+ */
+public class Parser {
+    private static final String COMMAND_EXIT = "bye";
+    private static final String COMMAND_LIST = "list";
+    private static final String COMMAND_DONE = "done";
+    private static final String COMMAND_TODO = "todo";
+    private static final String COMMAND_DEADLINE = "deadline";
+    private static final String COMMAND_EVENT = "event";
+    private static final String COMMAND_DELETE = "delete";
+    private static final String COMMAND_FIND = "find";
+
+    private static LocalDateTime toDateTime(String dateTime) throws InvalidDateException {
+        String[] dateTimeSplit = dateTime.split(" ");
+        if (dateTimeSplit.length != 2) {
+            throw new InvalidDateException("Your date needs to"
+                    + " have this format:\n\"YYYY-MM-DD HHMM\"");
+        }
+
+        String[] date = dateTimeSplit[0].split("-");
+        String time = dateTimeSplit[1];
+
+        if (date.length != 3) {
+            throw new InvalidDateException("Your date needs to"
+                    + " have this format:\n\"YYYY-MM-DD\"");
+        }
+        if (time.length() != 4) {
+            throw new InvalidDateException("Your time needs to"
+                    + " have this format:\n\"HHMM\"");
+        }
+
+        try {
+            int year = Integer.parseInt(date[0]);
+            int month = Integer.parseInt(date[1]);
+            int day = Integer.parseInt(date[2]);
+            int hour = Integer.parseInt(time.substring(0, 2));
+            int minute = Integer.parseInt(time.substring(2));
+
+            return LocalDateTime.of(year, month, day, hour, minute);
+        } catch (NumberFormatException | DateTimeException e) {
+            throw new InvalidDateException("Please check that you've"
+                    + " entered the date and time correctly");
+        }
+    }
+
+    private static Command parseAdd(String[] input)

```

Comment:
You can consider naming it as "parseCommand" instead of "parseAdd" to make it a bit clearer? 🤔

Reply:
The `parseAdd` method is meant for the parsing only commands to add new tasks, so `parseCommand` would be too broad of a name for this method. Admittedly, `parseAdd` isn't a very good name either, so I would be happy if you had more suggestions!